src/modules/AllWidgets.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+
+namespace WPML\PB\Elementor\Modules;
+
+
+class AllWidgets extends \WPML_Elementor_Media_Node {
+
+	/**
+	 * @param array $settings
+	 * @param string $target_lang
+	 * @param string $source_lang
+	 *
+	 * @return array
+	 */
+	public function translate( $settings, $target_lang, $source_lang ) {
+		global $sitepress;
+		$background = [ '_background_image', 'background_image', 'background_overlay_image', 'background_hover_image' ];
+		foreach ( $background as $element ) {
+			if ( isset( $settings[ $element ]["url"] ) ) {
+				$media_translate             = new \WPML_Media_Image_Translate( $sitepress,
+					new \WPML_Media_Attachment_By_URL_Factory() );
+				$settings[ $element ]["url"] = $media_translate->get_translated_image_by_url( $settings[ $element ]["url"],
+					$source_lang, $target_lang );
+			}
+		}
+
+		return $settings;
```

Comment:
You can simplify the whole method:

```php
$backgrounds = [
    'background_image',
    'background_overlay_image',
    'background_hover_image'
];

foreach ( $backgrounds as $background ) {
	$settings = $this->media_translate->translate_image_property( $settings, $background, $target_lang, $source_lang );
}

return $settings;
```

You don't need to instantiate `WPML_Media_Image_Translate` because it's already a dependency of this class.

Generally, we should instantiate classes only in factories. Otherwise it does not allow proper testing because we can't mock the object.

Reply:
I addressed that now.